core/src/main/scala/com/rallyhealth/vapors/factfilter/data/FactOrFactSet.scala

```diff
@@ -0,0 +1,20 @@
+package com.rallyhealth.vapors.factfilter.data
+
+import cats.Order
+import cats.instances.order._
+
+import scala.collection.immutable.SortedSet
+
+final class FactOrFactSet private[FactOrFactSet] (val toSortedSet: SortedSet[Fact]) extends AnyVal
```

Comment:
`FactOrFactSet` is probably asking for another name. Perhaps just `FactSet`?

Reply:
There already is a `FactSet`. This is magnet type. Its only purpose is to avoid method overloading for the `Evidence` and `FactSet` factory methods.

Reply:
I agree that there really needs to be better names in this.  I know that your development is a bit of a hydra-model right now, so I don't know how painful this will be to do in the middle.

But this will need to be done before too much longer.

Reply:
I think there may be some misunderstanding about this type. It should not be referenced directly. It only exists as a "magnet type", which is basically a target for implicit conversion that you can put on a method (can also be as varargs) that allows you to define various implicit conversions for different known types. This allows you to avoid method overloading, as well as allowing the implicit resolution system to prove type correctness before dropping the unnecessary type parameters (depending on what you keep in the magnet type). In my case, I'm just using this magnet type to allow calling the `Evidence.apply()` method with any variable combination of single facts, list of facts, set of facts, etc. without needing to define separate apply methods. In addition, this same magnet type is used in the `FactTable.apply` method to allow the same types of arguments to be passed.